tests/pruners_tests/test_successive_halving.py

```diff
@@ -0,0 +1,184 @@
+import optuna
+
+
+def test_successive_halving_pruner_with_one_trial():
+    # type: () -> None
+
+    study = optuna.study.create_study()
+    trial = optuna.trial.Trial(study, study.storage.create_new_trial_id(study.study_id))
+    trial.report(1, 1)
+    pruner = optuna.pruners.SuccessiveHalvingPruner(
+        min_resource=1, reduction_factor=2, min_early_stopping_rate=0)
+
+    # A pruner is not activated at a first trial.
+    assert not pruner.prune(study.storage, study.study_id, trial.trial_id, step=1)
```

Comment:
I could not find the case where the target value of the pruner is `NaN`. Please add a test function like `test_successive_halving_pruner_with_nan` to confirm trials with `NaN` values are pruned without any exceptions.